src/controllers/keyboard/keyboardcontrollerpreset.cpp

```diff
@@ -0,0 +1,242 @@
+        while (keyseqsRawI != keyseqsRaw.constBegin()) {
+            --keyseqsRawI;
+
+            // Check if this key sequence's target layout
+            // is the same as the user's language
+            if (keyseqsRawI->lang == layoutName) {
+                break;
+            }
+
+            // If none found, set keyseqsRawI to end(), which is
+            // kind of setting it to null
+            if (keyseqsRawI == keyseqsRaw.constBegin()) {
+                keyseqsRawI = keyseqsRaw.constEnd();
+                break;
+            }
+        }
+
+        // If no keyseq objects found for this layout, check if there
+        // is one that targets no specific layout. One whose 'lang'
+        // attribute was not given in the XML.
+        //
+        // NOTE: If there is more than one keyseq whose 'lang' attribute
+        //       is null, the last one will be chosen.
+        if (keyseqsRawI == keyseqsRaw.constEnd()) {
+            while (keyseqsRawI != keyseqsRaw.constBegin()) {
+                --keyseqsRawI;
+
+                // If lang is not given, that means that it is
+                // targeted at all not explicitly listed layouts
+                if (keyseqsRawI->lang.isEmpty()) {
+                    break;
+                }
+
+                // Same as in the first loop, setting to "null"-ish
+                // if none was found
+                if (keyseqsRawI == keyseqsRaw.constBegin()) {
+                    keyseqsRawI = keyseqsRaw.constEnd();
+                    break;
+                }
+            }
+        }
+
+        // If no keyseq was found for current layout, nor a global layout targeted at no
+        // layout in particular, Mixxx can't load in any mapping for this current action
+        if (keyseqsRawI == keyseqsRaw.constEnd()) {
+            qWarning() << "No keyseq found for" << layoutName
+                       << "nor any translatable keyseq. Skipping"
+                       << "mapping for ConfigKey" << configKey;
+            continue;
+        }
+
+        // Check if this keyseq is position based (if 'byPositionOf' attribute was given)
+        bool positionBased = !keyseqsRawI->byPositionOf.isEmpty();
+
+        // Check if this keyseq is position based, based on the current layout
+        bool byPositionOfCurrentLayout = keyseqsRawI->byPositionOf == layoutName;
+
+        // If 'byPositionOf' attribute not given, or given but where the
+        // layout is matches the current local, no need to translate
+        if (!positionBased || byPositionOfCurrentLayout) {
+            keyseqNeedsTranslate = false;
+        }
+
+        // Key sequences for control. If key sequence does not need any translation,
+        // this array will always contain just one key sequence. However, after
+        // translation it could potentially contain multiple key sequences.
+        QStringList keyseqs = QStringList() += (!keyseqsRaw.isEmpty() ? keyseqsRawI->keysequence : "");
```

Comment:
Please break into multiple lines. This is too complicated to read.

And the distinction between keyseqsRaw and keyseqsRawI is very subtle here, I don't like this naming.